### ISSUE TITLE
AttributeError: 'GeneratorContextManager' object has no attribute 'next'

### ISSUE THREAD

**Issue by yarikoptic** [pauldmccarthy/indexed_gzip#10]
Discovered while building updated package for NeuroDebian -- Have you encountered following issue on the older (eg. Debian stable with python 2.7.13 and Ubuntu 17.10 with python 2.7.14) systems (ok on newer with 2.7.15)?

```
_______________ test_IndexedGzipFile_pread_threaded_drop_handles _______________

testfile = '/build/indexed-gzip-0.8.5/ctest_zran_16777217_False.gz'
nelems = 16777217

    def test_IndexedGzipFile_pread_threaded_drop_handles(testfile, nelems):
>       _test_IndexedGzipFile_pread_threaded(testfile, nelems, True)

indexed_gzip/tests/test_indexed_gzip_threading.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
indexed_gzip/tests/test_indexed_gzip_threading.py:54: in _test_IndexedGzipFile_pread_threaded
    drop_handles=drop) as f:
indexed_gzip/indexed_gzip.pyx:831: in indexed_gzip.indexed_gzip.IndexedGzipFile.__init__ (indexed_gzip/indexed_gzip.c:11399)
    fobj            = _IndexedGzipFile(*args, **kwargs)
indexed_gzip/indexed_gzip.pyx:210: in indexed_gzip.indexed_gzip._IndexedGzipFile.__cinit__ (indexed_gzip/indexed_gzip.c:2587)
    with self.__file_handle():
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <contextlib.GeneratorContextManager object at 0x7fcdca983810>

    def __enter__(self):
        try:
>           return self.gen.next()
E           AttributeError: 'GeneratorContextManager' object has no attribute 'next'
```
(many other tests fail similarly).  Any immediate clues?

**Comment by pauldmccarthy** [comment]
Very strange ... it could be a cython issue - I just stumbled across cython/cython#1750, which has a comment that lists the same error.

I'm away from my desk right now, but wll look into this when I'm back on tuesday.

**Comment by yarikoptic** [comment]
Ah, that could be it since we indeed do not  provide cython backports in NeuroDebian... If there would be no easy way around this issue,  I will look into shipping "precompiled" pyx files for older cython

**Comment by pauldmccarthy** [comment]
@yarikoptic - I reproduced this on my mac (python 2.7.13+cython 0.25.2),  and tried the fix suggested at https://stackoverflow.com/a/44738561 (second part of that answer). It seems to work for me, and travis-CI is currently running the full test suite - mind giving it a go? If it works for you I'll get a bugfix release out this week.

The fix is on the `bf/cython_generatorcontextmanager_bug` branch: 1f4b230b09cef4bfe8812e9c750774890fd65c4b

**Comment by yarikoptic** [comment]
Confirming that it did resolve those build failures on my end.  I am ready to build final -2 revision for debian with the patch, or could wait for your release if you are planing to release some time soon with that change.  Just let me know

**Comment by pauldmccarthy** [comment]
I should be able to get an 0.8.6 release out tomorrow!

**Comment by yarikoptic** [comment]
cool, then I will just wait to not proliferate versions of the package ;-)  although it is all ready to be uploaded

**Comment by yarikoptic** [comment]
actually may be I will upload regardless to see if debian ci issue gets resolved -- may be there is more to it

;-)

**Comment by pauldmccarthy** [comment]
Fixed as of v0.8.6 - should be up on pypi in the next few hours!

**Comment by yarikoptic** [comment]
thanks, cheers!
